Kernel/Arch/x86/CPU.h

```diff
@@ -195,6 +195,50 @@ class PageDirectoryPointerTable {
 class GenericInterruptHandler;
 struct RegisterState;
 
+template<typename T>
+void read_possibly_unaligned_data(u8* where, T& data)
+{
+    // FIXME: Implement 64 bit unaligned data access
+    VERIFY(sizeof(T) == 8 || sizeof(T) == 4 || sizeof(T) == 2);
```

Comment:
Sorry, a little late to the party, but I think we should consider doing a `static_assert(sizeof(T) != 8)` here until it's implemented.

Reply:
But then you won't be able to access aligned 64 bit pointers, so the only thing you've got here is to disable all 64 bit accesses even though it should work just fine for many cases.